### PR TITLE
[FRONT-99] Show multiple lines in chart

### PR DESCRIPTION
**Pull request by DamianGrosz2**
### Description
- Updated the index of Compare so multiple lines are shown.
- Added a "Normalize data" button, so that the data all starts from the same day (as Viet wanted)
- Only when multiple lines are shown, the buttons "Select timeframe" and "Normalize data" are displayed
- Readded the functionality for "Select timeframe" button
- Disabled tooltip for multiple lines

### How I implemented

- Commented the "Select value" modal
- Added multiple colors

### Other
Bild 08 06 23 um 19 41

Bildschirm­foto 2023-06-08 um 19 42 22

### CODE REVIEW

File: src/components/page/details/Chart.tsx [hunk 1]
```diff
@@ -43,89 +68,148 @@ type ChartProps = {
...
+  const pastDate = now.setMonth(now.getMonth() - months)
+  return data.filter((d) => new Date(d.date).getTime() >= pastDate)
 }
 
-const Chart = ({ datasets }: ChartProps) => {
-  const [modalValue, setModalValue] = useState('Select Value')
-  const [isModalOpen, setIsModalOpen] = useState(false)
+const Chart = ({ datasets, multipleLines }: ChartProps) => {
+  // const [modalValue, setModalValue] = useState('Select Value')
+  // const [isModalOpen, setIsModalOpen] = useState(false)
```
Comment (ArnoClaude) [inline src/components/page/details/Chart.tsx:87]: Remove comments if not needed now

**Comment by ArnoClaude** [comment]
Looks really good, great work, Damian! The UX could definitely need some improvements, especially when there are a lot of projects/graphs in the figure.
Some suggestions:
- Sort the names of the project below the figure by their current (most recent) value. So, in this example, `gorilla` is first, then `chinese-poetry` would come next, because its line/current value is second
image
- When hovering over a graph, have a tooltip that shows the name of the repository you are hovering over (as well as the value on the point you are hovering over)

**Review by ArnoClaude — CHANGES_REQUESTED** [review]

**Comment by DamianGrosz2** [comment]
> * Sort the names of the project below the figure by their current (most recent) value. So, in this example, `gorilla` is first, then `chinese-poetry` would come next, because its line/current value is second



> Looks really good, great work, Damian! The UX could definitely need some improvements, especially when there are a lot of projects/graphs in the figure. Some suggestions:
> 
> * Sort the names of the project below the figure by their current (most recent) value. So, in this example, `gorilla` is first, then `chinese-poetry` would come next, because its line/current value is second
>   image
> * When hovering over a graph, have a tooltip that shows the name of the repository you are hovering over (as well as the value on the point you are hovering over)



> Looks really good, great work, Damian! The UX could definitely need some improvements, especially when there are a lot of projects/graphs in the figure. Some suggestions:
> 
> * Sort the names of the project below the figure by their current (most recent) value. So, in this example, `gorilla` is first, then `chinese-poetry` would come next, because its line/current value is second
>   image
> * When hovering over a graph, have a tooltip that shows the name of the repository you are hovering over (as well as the value on the point you are hovering over)

1. Done
2. Recharts doesn't provide this kind of tooltip for only one line, but rather all lines on a given x value.

**Review by ArnoClaude — APPROVED** [review]
Looks good to me now :)

File: src/components/page/details/Chart.tsx [hunk 2]
```diff
@@ -25,7 +25,32 @@ type ColorObject = {
...
 
+const singleColors = ['teal', 'red', 'mustard', 'yellow', 'orange', 'purple', 'blue', 'green']
 const grayColors = fullConfig.theme?.colors?.gray as ColorObject
+const indigoColors = fullConfig.theme?.colors?.indigo as ColorObject
+const colorValues = ['300', '500']
+
+const colors: string[] = []
+
+// Add multiple colors to the array
+colorValues.forEach((value) => {
```
Comment (pczern) [inline src/components/page/details/Chart.tsx:36]: use `map`

File: src/components/page/details/Chart.tsx [hunk 3]
```diff
@@ -25,7 +25,32 @@ type ColorObject = {
...
+  if (grayValue) {
+    colors.push(grayValue)
+  }
+  if (indigoValue) {
+    colors.push(indigoValue)
+  }
+})
+
+// Add single colors to the array
+singleColors.forEach((colorName) => {
```
Comment (pczern) [inline src/components/page/details/Chart.tsx:48]: use `concat`

File: src/components/page/details/Chart.tsx [hunk 4]
```diff
@@ -43,89 +68,112 @@ type ChartProps = {
...
-                ))}
-              </Modal>
+                <Modal isOpen={timeframeModalOpen} onClose={() => setTimeframeModalOpen(false)}>
+                  {TimeframeOptions.map((option) => (
+                    <Button
+                      key={option.label}
+                      variant="noBorderNoBG"
+                      text={option.label}
+                      fullWidth
+                      onClick={() => handleTimeframeChange(option.value)}
```
Comment (pczern) [inline src/components/page/details/Chart.tsx:162]: handlers are usually already wrapped, change this to onClick={handleTimeframeChange...

**Review by pczern — CHANGES_REQUESTED** [review]

File: src/components/page/details/Chart.tsx (hunk 2 above)
Comment (DamianGrosz2) [reply src/components/page/details/Chart.tsx:36]: done

File: src/components/page/details/Chart.tsx (hunk 3 above)
Comment (DamianGrosz2) [reply src/components/page/details/Chart.tsx:48]: done

File: src/components/page/details/Chart.tsx (hunk 4 above)
Comment (DamianGrosz2) [reply src/components/page/details/Chart.tsx:162]: done

**Review by mohammad-akkad — APPROVED** [review]

**Pull request merged**